x-pack/plugins/cases/server/client/client.ts

```diff
@@ -5,29 +5,40 @@
  * 2.0.
  */
 
-import { CasesClient, CasesClientArgs } from './types';
-import { createCasesSubClient } from './cases/client';
-import { createAttachmentsSubClient } from './attachments/client';
-import { createUserActionsSubClient } from './user_actions/client';
-import { createCasesClientInternal } from './client_internal';
+import { CasesClientArgs } from './types';
+import { CasesSubClient, createCasesSubClient } from './cases/client';
+import { AttachmentsSubClient, createAttachmentsSubClient } from './attachments/client';
+import { UserActionsSubClient, createUserActionsSubClient } from './user_actions/client';
+import { CasesClientInternal, createCasesClientInternal } from './client_internal';
 
-export const createCasesClient = (args: CasesClientArgs): CasesClient => {
-  const casesInternalClient = createCasesClientInternal(args, () => casesClient);
+export class CasesClient {
+  private readonly args: CasesClientArgs;
```

Comment:
Hmm do we need `args` as a member of the class?

Reply:
No, we don't 😄 . I am gonna fix it.